src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/ConfiguredAsyncDisposable.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+namespace System.Runtime.CompilerServices
```

Comment:
>System.Runtime.CompilerServices [](start = 10, length = 31)

why did we choose to put this type in the System.Runtime.CompilerServices namespace? I understand the compiler will use it but still why not used System.Threading.Tasks? it looks to be more reasonable to be there.

Reply:
> why did we choose to put this type in the System.Runtime.CompilerServices namespace? I understand the compiler will use it

Because this is where all of the other "you pretty much never refer to this type in your own code and instead the compiler gets it as part of some operation" types lives, e.g. when you "await task.ConfigureAwait(false)", the type returned from ConfigureAwait is a ConfiguredTaskAwaitable that lives in System.Runtime.CompilerServices.

Reply:
I noticed the constructor is internal, so now I understand might be this is the reason we have it System.Runtime.CompilerServices namespace as it is not expected users will get construct it directly. maybe adding some comment regarding that would be useful.

Reply:
I posted last comment before I read your last reply. do we need to hide this type from the intellisense then?

Reply:
We don't the other "Configured..." types, or in general the other types in System.Runtime.CompilerServices.  In general if it's in System.Runtime.CompilerServices, it's assumed it's compiler-focused.